---
title: 编译器错误 CS1015
ms.date: 07/20/2015
f1_keywords:
- CS1015
helpviewer_keywords:
- CS1015
ms.assetid: 53179feb-e8be-41e0-bb0b-f7879e9fa613
ms.openlocfilehash: 763b8b8e0aaf26b8c7813051e766d65f28411955
ms.sourcegitcommit: 9b552addadfb57fab0b9e7852ed4f1f1b8a42f8e
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 04/23/2019
ms.locfileid: "61654088"
---
# <a name="compiler-error-cs1015"></a><span data-ttu-id="023f4-102">编译器错误 CS1015</span><span class="sxs-lookup"><span data-stu-id="023f4-102">Compiler Error CS1015</span></span>
<span data-ttu-id="023f4-103">应是对象、字符串或类类型</span><span class="sxs-lookup"><span data-stu-id="023f4-103">An object, string, or class type expected</span></span>  
  
 <span data-ttu-id="023f4-104">尝试将预定义的数据类型传入 [catch](../../csharp/language-reference/keywords/try-catch.md) 块。</span><span class="sxs-lookup"><span data-stu-id="023f4-104">An attempt was made to pass a predefined data type into a [catch](../../csharp/language-reference/keywords/try-catch.md) block.</span></span> <span data-ttu-id="023f4-105">仅派生自 <xref:System.Exception?displayProperty=nameWithType> 的数据类型可传入 `catch` 块。</span><span class="sxs-lookup"><span data-stu-id="023f4-105">Only data types that derive from <xref:System.Exception?displayProperty=nameWithType> can be passed into a `catch` block.</span></span> <span data-ttu-id="023f4-106">有关异常的详细信息，请参阅 [Exception Handling Statements](../../csharp/language-reference/keywords/exception-handling-statements.md)。</span><span class="sxs-lookup"><span data-stu-id="023f4-106">For more information on exceptions, see [Exception Handling Statements](../../csharp/language-reference/keywords/exception-handling-statements.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="023f4-107">示例</span><span class="sxs-lookup"><span data-stu-id="023f4-107">Example</span></span>  
 <span data-ttu-id="023f4-108">下面的示例生成 CS1015：</span><span class="sxs-lookup"><span data-stu-id="023f4-108">The following sample generates CS1015:</span></span>  
  
```csharp  
// CS1015.cs  
class Sample  
{  
    static void Main()  
    {  
        try   
        {  
        }  
        catch(int)   // CS1015, int is not derived from System.Exception  
        {  
        }  
    }  
}  
```
